### PR TITLE
Leaders/season stat grid

### PR DESCRIPTION
**Pull request by JacobLee23**
Parser for the FanGraphs [Season Stat Grid](https://fangraphs.com/season-stat-grid)

```python
from FanGraphs import leaders
ssg = leaders.SeasonStatGrid()
```

<h2>Methods</h2>

- `list_queries(self)`
- `list_options(self, query)`
- `current_option(self, query)`
- `configure(query, option)`
- `quit(self)`
- `reset(self)`
- `export(self, name="", *, size="Infinity", sortby="Name", reverse=False)`

### CODE REVIEW

File: functional_tests.py [hunk 1]
```diff
@@ -1,14 +1,41 @@
...
 import os
 import random
 import unittest
 from urllib.request import urlopen
 
+from FanGraphs import exceptions
 from FanGraphs import leaders
 
 
+@unittest.SkipTest
```
Comment (JacobLee23) [inline functional_tests.py:14]: Remove `unittest.SkipTest` decorator

File: functional_tests.py [hunk 2]
```diff
@@ -1,14 +1,41 @@
...
+            exceptions.InvalidFilterQuery
+        ):
+            parser.current_option("nonexistent query")
+
+        with self.assertRaises(
+            exceptions.InvalidFilterQuery
+        ):
+            parser.configure("nonexistent query", "nonexistent option")
+
+        parser.quit()
```
Comment (JacobLee23) [inline functional_tests.py:35]: Add exception tests for `SeasonStatGrid` class.

File: functional_tests.py [hunk 3]
```diff
@@ -1,14 +1,41 @@
...
+
+
+@unittest.SkipTest
```
Comment (JLpython-py) [inline functional_tests.py:38]: Remove `unittest.SkipTest` decorator.

File: functional_tests.py [hunk 4]
```diff
@@ -125,7 +137,101 @@ def test_reset(self):
...
+    @classmethod
+    def setUpClass(cls):
+        cls.base_url = cls.parser.browser.current_url
+
+    @classmethod
+    def tearDownClass(cls):
+        for file in os.listdir("out"):
+            os.remove(os.path.join("out", file))
+        os.rmdir("out")
+        cls.parser.quit()
```
Comment (JacobLee23) [inline functional_tests.py:157]: `setUpClass` and `tearDownClass` are unnecessary if `TestSeasonStatGrid` has only one instance method. It is preferable that the tests are split up by method/functionality.

File: requirements.txt [hunk 5]
```diff
@@ -1,3 +1,10 @@
+beautifulsoup4==4.9.3
+certifi==2020.12.5
+chardet==4.0.0
+cssselect==1.1.0
+idna==2.10
 lxml==4.6.2
+requests==2.25.1
```
Comment (JacobLee23) [inline requirements.txt:7]: The `requests` package is not necessary, as of now.

File: functional_tests.py (hunk 3 above)
Comment (JacobLee23) [inline functional_tests.py:38]: Remove `unittest.SkipTest` decorator.

**Pull request closed without merging**